lib/Parser/ptree.h

```diff
@@ -856,55 +919,33 @@ class ParseNodeSpecialName : public ParseNodePid
+    ParseNodeSuperReference(OpCode nop, charcount_t ichMin, charcount_t ichLim, ParseNode * pnode1, ParseNode * pnode2);
+
     ParseNodePtr pnodeThis;
 };
 
 // call node with super target
 class ParseNodeSuperCall : public ParseNodeCall
 {
 public:
+    ParseNodeSuperCall(OpCode nop, charcount_t ichMin, charcount_t ichLim, ParseNode * pnodeTarget, ParseNode * pnodeArgs);
+
     ParseNodePtr pnodeThis;
     ParseNodePtr pnodeNewTarget;
+};
+
+#define AssertNodeMem(pnode) AssertPvCb(pnode, sizeof(ParseNode))
+#define AssertNodeMemN(pnode) AssertPvCbN(pnode, sizeof(ParseNode))
+
 
-    void Init(OpCode nop, ParseNodePtr pnodeTarget, ParseNodePtr pnodeArgs, charcount_t ichMin, charcount_t ichLim);
-};
-
-const int kcbPnNone           = sizeof(ParseNode);
-const int kcbPnArrLit         = sizeof(ParseNodeArrLit);
-const int kcbPnBin            = sizeof(ParseNodeBin);
-const int kcbPnBlock          = sizeof(ParseNodeBlock);
-const int kcbPnCall           = sizeof(ParseNodeCall);
-const int kcbPnCase           = sizeof(ParseNodeCase);
-const int kcbPnCatch          = sizeof(ParseNodeCatch);
-const int kcbPnClass          = sizeof(ParseNodeClass);
-const int kcbPnExportDefault  = sizeof(ParseNodeExportDefault);
-const int kcbPnFinally        = sizeof(ParseNodeFinally);
-const int kcbPnFlt            = sizeof(ParseNodeFloat);
-const int kcbPnFnc            = sizeof(ParseNodeFnc);
-const int kcbPnFor            = sizeof(ParseNodeFor);
-const int kcbPnForIn          = sizeof(ParseNodeForInOrForOf);
-const int kcbPnForOf          = sizeof(ParseNodeForInOrForOf);
-const int kcbPnIf             = sizeof(ParseNodeIf);
-const int kcbPnInt            = sizeof(ParseNodeInt);
-const int kcbPnJump           = sizeof(ParseNodeJump);
-const int kcbPnModule         = sizeof(ParseNodeModule);
-const int kcbPnParamPattern   = sizeof(ParseNodeParamPattern);
-const int kcbPnPid            = sizeof(ParseNodePid);
-const int kcbPnProg           = sizeof(ParseNodeProg);
-const int kcbPnReturn         = sizeof(ParseNodeReturn);
-const int kcbPnSpecialName    = sizeof(ParseNodeSpecialName);
-const int kcbPnStrTemplate    = sizeof(ParseNodeStrTemplate);
-const int kcbPnSuperCall      = sizeof(ParseNodeSuperCall);
-const int kcbPnSuperReference = sizeof(ParseNodeSuperReference);
-const int kcbPnSwitch         = sizeof(ParseNodeSwitch);
-const int kcbPnTri            = sizeof(ParseNodeTri);
-const int kcbPnTry            = sizeof(ParseNodeTry);
-const int kcbPnTryCatch       = sizeof(ParseNodeTryCatch);
-const int kcbPnTryFinally     = sizeof(ParseNodeTryFinally);
-const int kcbPnUni            = sizeof(ParseNodeUni);
-const int kcbPnVar            = sizeof(ParseNodeVar);
-const int kcbPnWhile          = sizeof(ParseNodeWhile);
-const int kcbPnWith           = sizeof(ParseNodeWith);
-
-#define AssertNodeMem(pnode) AssertPvCb(pnode, kcbPnNone)
-#define AssertNodeMemN(pnode) AssertPvCbN(pnode, kcbPnNone)
+typedef ParseNode ParseNodeNone;
+template <OpCode nop> class OpCodeTrait;
```

Comment:
I don't see ParseNodeNone used anywhere in this change. Why do we need this typedef?